File: C++/144A. Arrival of the General.cpp
#include<bits/stdc++.h>
using namespace std;
int main()
{
    int n;
    cin>>n;
    int b=0;
    vector<int> a, c;
    for(int i=0; i<n; i++){
        int lmao;
        cin>>lmao;
        a.push_back(lmao);
        c.push_back(lmao);
    }
    sort(a.begin(), a.end());
    for(int j=0; j<n; j++){
            if(c[j]==a[n-1]){
                c.insert(c.begin(), a[n-1]);
                c.erase(c.begin()+j+1);
                break;
            }
            else{
                b++;
            }
        }
    for(int j=n-1; j>0; j--){
            if(c[j]==a[0]){
              break;
            }
            else{
                b++;
            }
    }
    cout<<b<<endl;
    return 0;
}

File: Python/144A. Arrival of the General.py
def main():
    n = int(input())
    b = 0
    a = list(map(int, input().split()))
    c = []
    for i in range(n):
        c.append(a[i])
    a.sort()
    i = 0
    while i<n:
        if c[i] == a[n-1]:
            c.insert(0, c[i])
            c.pop(i+1)
            break
        else:
            b += 1
        i += 1
    i = n-1
    while i>=0:
        if c[i]==a[0]:
            break
        else:
            b += 1
        i -= 1
    print(b, "\n")
main()